### PR TITLE
Feature/spline 671 producer api 06

### PR DESCRIPTION
**Pull request by wajda**
Introduces necessary abstractions to generate Swagger defs, and a temporary conversion between v1 and v1.1 models to maintain the working app state. Further model upgrade will be done in #677

### CODE REVIEW

File: producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/v1/package.scala [hunk 1]
```diff
@@ -14,10 +14,10 @@
...
-package za.co.absa.spline.swagger
+package za.co.absa.spline.producer.rest.controller
 
-import java.io.File
+package object v1 {
 
-case class SwaggerDocGenConfig(
-  maybeOutputFile: Option[File] = None,
-  restContextClass: Option[Class[_]] = None)
+  trait _package
```
Comment (cerveada) [inline producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/v1/package.scala:21]: What is the purpose of this trait?

File: producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  other: Seq[DataOperation] = Nil
+) {
+  def all: Seq[OperationLike] = reads ++ other :+ write
+}
+
+case class Schemas(
+  attributes: Seq[AttributeReference],
+  functions: Seq[FunctionalExpression],
+  constants: Seq[Literal],
+  mapping: Map[OperationLike.Id, Array[ExpressionLike.Id]]
```
[the remark this answers is not in the document]
Comment (cerveada) [reply producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala:47]: According tu documentation is output. Do we need to have more than one output schema? It seems to me that output of operation represents a dataFrame and dataFrame should have just one schema, or not?

[the remark this answers is not in the document]
Comment (wajda) [reply producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala:47]: > Is this output or input of the operation?

Output

[the remark this answers is not in the document]
Comment (wajda) [reply producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala:47]: > According tu documentation is output. Do we need to have more than one output schema? It seems to me that output of operation represents a dataFrame and dataFrame should have just one schema, or not?

No, why? One output schema per operation, by definition.

File: producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/expressions.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  override val id: Id,
+  override val dataType: Option[DataType],
+  override val childIds: Seq[ExpressionLike.Id],
+  override val extra: Map[String, Any],
+  name: String,
+  args: Seq[ExpressionLike.Id],
+  params: Map[String, Any],
+) extends ExpressionLike
+
+case class AttributeReference(
```
[the remark this answers is not in the document]
Comment (wajda) [reply producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/expressions.scala:42]: renamed

File: producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/v1/package.scala (hunk 1 above)
Comment (wajda) [reply producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/v1/package.scala:21]: It's just an anchor type that is use to reference the package

File: producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (cerveada) [reply producer-model/src/main/scala/za/co/absa/spline/producer/model/v1_1/executionPlan.scala:47]: Sorry, my bad. I got confused. It's OK

**Pull request merged**